### ISSUE TITLE
[1.10.0] waitFor is broken

### ISSUE THREAD

**Issue by gorhom** [software-mansion/react-native-gesture-handler#1354]
## Description

With the new release of `v1.10.0` the prop `waitFor` is not working anymore. I did investigate a bit the commits that shipped with this release and it seems this commit https://github.com/software-mansion/react-native-gesture-handler/commit/8d679249604f0b6f69d3d3933a6383145fb7617a caused the issue.

### Screenshots

`v1.10.0`

https://user-images.githubusercontent.com/4061838/107699230-12314a80-6cb6-11eb-8d1c-31e8eb185718.mov

`v1.9.0`

https://user-images.githubusercontent.com/4061838/107699285-29703800-6cb6-11eb-91d7-79b147d19185.mov

## Steps To Reproduce

1. Create a `PanGestureHandler` that contain a `NativeGestureHandler` & any scrollable
2. Pass `PanGestureHandler` ref to `NativeGestureHandler` in `waitFor` prop

### Expected behavior

- `NativeGestureHandler` should block all events and wait for `PanGestureHandler` to finish

### Actual behavior

- `NativeGestureHandler` ignores passed `waitFor` ref

## Snack or minimal code example

```tsx
import React, {useRef} from 'react';
import {StyleSheet, View, ScrollView, Text} from 'react-native';
import {
  NativeViewGestureHandler,
  PanGestureHandler,
} from 'react-native-gesture-handler';

const App = () => {
  const panRef = useRef<PanGestureHandler>(null);
  return (
    <View style={styles.container}>
      <PanGestureHandler ref={panRef}>
        <NativeViewGestureHandler waitFor={panRef}>
          <ScrollView style={styles.container}>
            {Array(50)
              .fill(0)
              .map((_, index) => (
                <Text
                  key={`item-${index}`}
                  style={styles.item}>{`item-${index}`}</Text>
              ))}
          </ScrollView>
        </NativeViewGestureHandler>
      </PanGestureHandler>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  item: {
    padding: 6,
    margin: 3,
    backgroundColor: '#999',
  },
});
export default App;

```

## Package versions

- React: 16.13.1
- React Native: 0.63.4
- React Native Gesture Handler: 1.10.0

**Comment by jakub-gonet** [comment]
Thanks for the detailed description.
The fix (in #1357) should be merged and released shortly.

**Comment by gorhom** [comment]
Thanks @jakub-gonet 👏
